### PR TITLE
Fix issues: 48, 49, 50, 53, 54, 57

### PR DESCRIPTION
**Pull request by stefanb2**
I hope this cumulative pull request is OK

BTW: the first commit adds my name to the CONTRIBUTORS file

### CODE REVIEW

**Comment by stefanb2** [comment]
I was added to our companys' corporate CLA a few weeks back for AOSP contributions:

* email: stefanb@gpartner-nvidia.com
* organization: Nvidia

I was under the impression that this would be enough to submit to GitHub too.

**Comment by stefanb2** [comment]
I verified that I've added the email address stefanb@gpartner-nvidia.com to my GitHub account

**Comment by shinh** [comment]
Great! Thanks for handling the CLA stuff. These changes look great, though I'd like to see tests in the testcase directory. You gave me clear test cases so I'll merge this PR and add tests by myself. I think I'll do a few follow-up changes, too. Thanks again!

File: symtab.cc [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 #include "symtab.h"
 
+#include <iostream>
```
Comment (shinh) [inline symtab.cc]: Unnecessary

File: func.cc [hunk 2]
```diff
@@ -583,7 +583,7 @@ void CallFunc(const vector<Value*>& args, Evaluator* ev, string* s) {
     Intern("5"), Intern("6"),  Intern("7"), Intern("8"), Intern("9")
   };
 
-  const string&& func_name = args[0]->Eval(ev);
+  const string&& func_name = TrimSpace(args[0]->Eval(ev)).as_string();
```
Comment (shinh) [inline func.cc]: For ckati, I'm pathologically trying to avoid string allocations. Let me remove as_string() call.

File: flags.h [hunk 3]
```diff
@@ -34,6 +34,7 @@ struct Flags {
   bool is_dry_run;
   bool is_silent_mode;
   bool is_syntax_check_only;
+  bool posix_shell;
```
Comment (shinh) [inline flags.h]: This global state should be handled at evaluation time and it seems $(shell) should use this. I've moved this from here to eval.h. https://github.com/google/kati/commit/2941ea07651737a6151b19db45c010d2083dfa1d

**Pull request merged**